packages/js/src/dashboard/widgets/search-ranking-compare-widget.js

```diff
@@ -0,0 +1,196 @@
+ */
+export const createIcsaDataFormatter = ( dataTransformer, dataFormatter ) => ( initData ) => {
+	const data = dataTransformer( initData );
+	if ( Object.keys( data ).length === 2 ) {
+		return {
+			impressions: dataFormatter.format( data.impressions, "impressions" ),
+			clicks: dataFormatter.format( data.clicks, "clicks" ),
+		};
+	}
+	return {
+		impressions: dataFormatter.format( data.impressions, "impressions" ),
+		clicks: dataFormatter.format( data.clicks, "clicks" ),
+		ctr: dataFormatter.format( data.ctr, "ctr" ),
+		position: dataFormatter.format( data.position, "position" ),
+	};
+};
+
+/**
+ * @param {import("../services/data-provider")} dataProvider The data provider.
+ * @param {import("../services/remote-data-provider")} remoteDataProvider The remote data provider.
+ * @param {import("../services/comparison-metrics-data-formatter")} dataFormatter The data formatter.
+ * @returns {JSX.Element} The element.
+ */
+export const SearchRankingCompareWidget = ( { dataProvider, remoteDataProvider, dataFormatter } ) => {
+	/**
+	 * @param {RequestInit} options The options.
+	 * @returns {Promise<TimeBasedData|Error>} The promise of IcsaData or an Error.
+	 */
+	const getData = useCallback( ( options ) => {
+		return remoteDataProvider.fetchJson(
+			dataProvider.getEndpoint( "timeBasedSeoMetrics" ),
+			{ options: { widget: "searchRankingCompare" }  },
+			options );
+	}, [ dataProvider ] );
+
+	/**
+	 * @type {function(?TimeBasedData[]): SearchRankingCompareData} Function to format the widget data.
+	 * */
+	const formatIcsaData = useMemo( () => createIcsaDataFormatter( transformData, dataFormatter ), [ transformData, dataFormatter ] );
+
+	const { data, error, isPending } = useRemoteData( getData, formatIcsaData );
+
+	return <Widget className="yst-paper__content yst-col-span-4">
+		{ isPending && <SearchRankingCompareSkeletonLoader /> }
+		{ error && <Alert variant="error" className="yst-mt-4">{ error.message }</Alert> }
+		{ data && Object.keys( data ).length > 0 &&
+			<div className="yst-flex yst-flex-col yst-justify-center yst-items-center  @6xl:yst-flex-row @6xl:yst-justify-evenly rtl:yst-flex-row-reverse ">
+				<SearchRankingCompareMetric
+					metricName="Impressions"
+					data={ data.impressions }
+					tooltipLocalizedString={ __( "The number of times your website appeared in Google search results over the last 28 days.", "wordpress-seo" ) }
+				/>
+				<div className="yst-h-px yst-w-full yst-bg-slate-200 yst-my-6 @6xl:yst-h-20 @6xl:yst-w-px @6xl:yst-my-1" />
+				<SearchRankingCompareMetric
+					metricName="Clicks"
+					data={ data.clicks }
+					tooltipLocalizedString={ __( "The total number of times users clicked on your website's link in Google search results over the last 28 days.", "wordpress-seo" ) }
+				/>
+				<div className=" yst-h-px yst-w-full yst-bg-slate-200 yst-my-6 @6xl:yst-h-20 @6xl:yst-w-px @6xl:yst-my-1" />
+				<SearchRankingCompareMetric
+					metricName="CTR"
+					data={ data?.ctr ?? null }
```

Comment:
Ah, here are you `null` fallbacks. On a per metric basis 😅
This seems a bit strange to me.
You could have it already in the formatter and just always provide all 4 keys there.